src/spirc.rs

```diff
@@ -277,28 +328,25 @@ impl SpircTask {
 
             MessageType::kMessageTypePlay => {
                 if self.state.get_status() == PlayStatus::kPlayStatusPause {
-                    self.mixer.start();
-                    self.player.play();
-                    self.state.set_status(PlayStatus::kPlayStatusPlay);
-                    self.state.set_position_measured_at(now_ms() as u64);
+                    self.handle_play();
                 }
 
                 self.notify(None);
             }
 
-            MessageType::kMessageTypePause => {
+            MessageType::kMessageTypePlayPause => {
                 if self.state.get_status() == PlayStatus::kPlayStatusPlay {
-                    self.player.pause();
-                    self.mixer.stop();
-                    self.state.set_status(PlayStatus::kPlayStatusPause);
-
-                    let now = now_ms() as u64;
-                    let position = self.state.get_position_ms();
+                    self.handle_pause();
+                } else {
```

Comment:
That should be `else if self.state.get_status() == PlayStatus::kPlayStatusPause`